src/rules/transition/out.jl

```diff
@@ -1,27 +1,33 @@
 
-@rule Transition(:out, Marginalisation) (q_in::Categorical, q_a::MatrixDirichlet) = begin
+# Belief Propagation                #
+# --------------------------------- #
+
+@rule Transition(:out, Marginalisation) (m_in::Categorical, m_a::PointMass) = begin
+    @logscale 0
+    p = mean(m_a) * probvec(m_in)
+    normalize!(p, 1)
+    return Categorical(p)
+end
+
+@rule Transition(:out, Marginalisation) (q_in::PointMass, q_a::PointMass) = begin
+    p = mean(q_a) * mean(q_in)
+    normalize!(p, 1)
+    return Categorical(p)
+end
+
+# Variational                       # 
+# --------------------------------- #
+
+@rule Transition(:out, Marginalisation) (q_in::Categorical, q_a::Any) = begin
     a = clamp.(exp.(mean(log, q_a) * probvec(q_in)), tiny, Inf)
     return Categorical(a ./ sum(a))
 end
 
```

Comment:
Note that in this rule you're quoting the exponentiation operator closes after multiplication with q_in, so it wouldn't necessarily hold up, right?

Reply:
And, also, I discussed this with @bvdmitri and @ThijsvdLaar, and the conclusion was that it should be illegal for the user to specify a transition matrix with negative entries

Reply:
> Note that in this rule you're quoting the exponentiation operator closes after multiplication with q_in, so it wouldn't necessarily hold up, right?

I was hoping for some different rearrangements of the terms while deriving VMP message, such that you wouldn't have to take an element-wise log of the matrix. I will have a closer look. Perhaps, you are right. 

> And, also, I discussed this with @bvdmitri and @ThijsvdLaar, and the conclusion was that it should be illegal for the user to specify a transition matrix with negative entries

Perhaps that's true, given the variational constraints, i.e. when the user wants to perform VMP around the node. But the transition matrix can have negative entries.

Reply:
Indeed, @wouterwln. The way the node's function is expressed, VMP with a clamped matrix with non-positive elements, should be forbidden. Maybe throwing an error explaining that the user should change his/her matrix or use a BP rule instead will be nice.

Reply:
In the current implementation, it will throw an error anyway when executing the `log` on a non-positive elements, right?

Reply:
I'm a bit hesitant to implement a check, since I think it will be a significant computational overhead to check for something that will error in the same line anyway. @bvdmitri I don't think there's a way to augment an error message without computational overhead? Right?